### DISCUSSION TITLE
best way to use type UseMutationResult and UseQueryResult

### DISCUSSION THREAD

**Discussion by ShaneZhengNZ** [TanStack/query#6883] [Q&A]
I am trying to create my own wrapper around useMutation and useQuery, just so that I can do some common error handling, etc. 

In this case, I have to import type UseMutationResult and UseQueryResult from "@tanstack/react-query".

but, I get the following warning:

transform[stderr]: The exported identifier "UseMutationResult" is not declared in Babel's scope tracker
transform[stderr]: as a JavaScript value binding, and "@babel/plugin-transform-typescript"
transform[stderr]: never encountered it as a TypeScript type declaration.
transform[stderr]: It will be treated as a JavaScript value.
transform[stderr]:
transform[stderr]: This problem is likely caused by another plugin injecting
transform[stderr]: "UseMutationResult" without registering it in the scope tracker. If you are the author
transform[stderr]:  of that plugin, please use "scope.registerDeclaration(declarationPath)".
transform[stderr]: The exported identifier "UseQueryResult" is not declared in Babel's scope tracker
transform[stderr]: as a JavaScript value binding, and "@babel/plugin-transform-typescript"
transform[stderr]: never encountered it as a TypeScript type declaration.
transform[stderr]: It will be treated as a JavaScript value.
transform[stderr]:
transform[stderr]: This problem is likely caused by another plugin injecting
transform[stderr]: "UseQueryResult" without registering it in the scope tracker. If you are the author
transform[stderr]:  of that plugin, please use "scope.registerDeclaration(declarationPath)".

any suggested fixes?

**Comment by ryo-kozin** [comment]
@ShaneZhengNZ 
Have you tried explicitly using `import type` for `UseMutationResult` and `UseQueryResult`?

For example:
```

import type { UseMutationResult, UseQueryResult } from "@tanstack/react-query";

```

That should tell Babel/TypeScript to strip the import entirely and silence the warning.
